plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/metastore/DeltaLakeTableMetadataScheduler.java

```diff
@@ -0,0 +1,258 @@
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+import static com.google.common.util.concurrent.MoreExecutors.newDirectExecutorService;
+import static io.airlift.concurrent.Threads.daemonThreadsNamed;
+import static io.airlift.concurrent.Threads.threadsNamed;
+import static io.trino.plugin.deltalake.DeltaLakeSessionProperties.storeTableMetadata;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.ColumnMappingMode.NONE;
+import static io.trino.plugin.deltalake.transactionlog.DeltaLakeSchemaSupport.getColumnMetadata;
+import static io.trino.plugin.hive.HiveMetadata.TABLE_COMMENT;
+import static java.util.Comparator.comparing;
+import static java.util.Objects.requireNonNull;
+import static java.util.concurrent.Executors.newFixedThreadPool;
+import static java.util.concurrent.Executors.newSingleThreadScheduledExecutor;
+import static java.util.concurrent.TimeUnit.MILLISECONDS;
+import static java.util.function.BinaryOperator.maxBy;
+
+public class DeltaLakeTableMetadataScheduler
+{
+    private static final Logger log = Logger.get(DeltaLakeTableMetadataScheduler.class);
+
+    private static final String TRINO_LAST_TRANSACTION_VERSION = "trino_last_transaction_version";
+    private static final String TRINO_METADATA_SCHEMA_STRING = "trino_metadata_schema_string";
+    private static final int MAX_FAILED_COUNTS = 10;
+
+    private final DeltaLakeTableOperationsProvider tableOperationsProvider;
+    private final TypeManager typeManager;
+    private final int tableParameterLengthLimit;
+    private final int storeTableMetadataThreads;
+    private final Map<SchemaTableName, TableUpdateInfo> updateInfos = new ConcurrentHashMap<>();
+    private final boolean enabled;
+
+    private ExecutorService executor;
+    private ScheduledExecutorService scheduler;
+    private final AtomicInteger failedCounts = new AtomicInteger();
+
+    @Inject
+    public DeltaLakeTableMetadataScheduler(
+            DeltaLakeTableOperationsProvider tableOperationsProvider,
+            NodeManager nodeManager,
+            TypeManager typeManager,
+            @TableParameterLengthLimit int tableParameterLengthLimit,
+            DeltaLakeConfig config)
+    {
+        this.tableOperationsProvider = requireNonNull(tableOperationsProvider, "tableOperationsProvider is null");
+        this.typeManager = requireNonNull(typeManager, "typeManager is null");
+        this.tableParameterLengthLimit = tableParameterLengthLimit;
+        this.storeTableMetadataThreads = config.getStoreTableMetadataThreads();
+        requireNonNull(nodeManager, "nodeManager is null");
+        this.enabled = config.isStoreTableMetadataEnabled() && nodeManager.getCurrentNode().isCoordinator();
+    }
+
+    @Managed
+    public boolean isShutdown()
+    {
+        return scheduler.isShutdown();
+    }
+
+    public void putAll(Map<SchemaTableName, TableUpdateInfo> tableParameters)
+    {
+        updateInfos.putAll(tableParameters);
```

Comment:
This should be a no-op if the scheduler is stopped.

Please add also a corresponding test to simulate continuos failures which would ultimately lead to shutting down the scheduler and afterwards to still provide a good quality of service for performing metadata (columns & comments) queries.

Reply:
I'm thinking about `io.trino.plugin.deltalake.metastore.glue.TestDeltaLakeConcurrentModificationGlueMetastore` to test a fake AccessDenied case - maybe too complicated for the task at hand.